core/chains/evm/txmgr/orm.go

```diff
@@ -754,3 +776,171 @@ func (o *orm) SaveInProgressAttempt(attempt *EthTxAttempt) error {
 	}
 	return nil
 }
+
+// FindEthTxsRequiringGasBump returns transactions that have all
+// attempts which are unconfirmed for at least gasBumpThreshold blocks,
+// limited by limit pending transactions
+//
+// It also returns eth_txes that are unconfirmed with no eth_tx_attempts
+func (o *orm) FindEthTxsRequiringGasBump(ctx context.Context, address common.Address, blockNum, gasBumpThreshold, depth int64, chainID big.Int) (etxs []*EthTx, err error) {
+	if gasBumpThreshold == 0 {
+		return
+	}
+	qq := o.q.WithOpts(pg.WithParentCtx(ctx))
+	err = qq.Transaction(func(tx pg.Queryer) error {
+		stmt := `
+SELECT eth_txes.* FROM eth_txes
+LEFT JOIN eth_tx_attempts ON eth_txes.id = eth_tx_attempts.eth_tx_id AND (broadcast_before_block_num > $4 OR broadcast_before_block_num IS NULL OR eth_tx_attempts.state != 'broadcast')
+WHERE eth_txes.state = 'unconfirmed' AND eth_tx_attempts.id IS NULL AND eth_txes.from_address = $1 AND eth_txes.evm_chain_id = $2
+	AND (($3 = 0) OR (eth_txes.id IN (SELECT id FROM eth_txes WHERE state = 'unconfirmed' AND from_address = $1 ORDER BY nonce ASC LIMIT $3)))
+ORDER BY nonce ASC
+`
+		if err = tx.Select(&etxs, stmt, address, chainID.String(), depth, blockNum-gasBumpThreshold); err != nil {
+			return errors.Wrap(err, "FindEthTxsRequiringGasBump failed to load eth_txes")
+		}
+		err = o.LoadEthTxesAttempts(etxs, pg.WithQueryer(tx))
+		return errors.Wrap(err, "FindEthTxsRequiringGasBump failed to load eth_tx_attempts")
+	}, pg.OptReadOnlyTx())
+	return
+}
+
+// FindEthTxsRequiringResubmissionDueToInsufficientEth returns transactions
+// that need to be re-sent because they hit an out-of-eth error on a previous
+// block
+func (o *orm) FindEthTxsRequiringResubmissionDueToInsufficientEth(address common.Address, chainID big.Int, qopts ...pg.QOpt) (etxs []*EthTx, err error) {
+	qq := o.q.WithOpts(qopts...)
+	// qq := q.WithOpts(pg.WithParentCtx(ctx))
```

Comment:
Why does this not need the parent ctx?

Reply:
Oh yeah, I should remove that comment. 

You don't need parent context anymore because we accept qopts so we can pass in the parent context as one of the side effects of the pg.QOpt function. In fact that's what we do here https://github.com/smartcontractkit/chainlink/pull/8543/files#diff-d6b76588977cc18e5b4c043aaae86416044e9b7e4233e6a79f5b05f30aa5219fR655

Reply:
If it's ok i'll remove comment in next PR just so that I don't need to wait for another round of CI tests

Reply:
Sounds good. Also, could we stick to this pattern at all places in this file? Some functions still have the "q.WithOpts(pg.WithParentCtx(ctx))" usage.

Reply:
Yeah, that's what this ticket is for https://smartcontract-it.atlassian.net/browse/BCI-917 . We decided it was out of scope for this ticket so we pulled it out.